### PR TITLE
Move database access code out of ``galaxy.util``

### PR DESCRIPTION
**Pull request by jdavcs**
This removes database access code from the `galaxy.util` package. Package tests currently pass because such code is invoked indirectly, through `trans.sa_session` with `trans` passed as an argument. However, refactoring towards SA 2.0 (replacing the legacy `query` object with `select`) exposes this and tests fail (we have to `from sqlalchemy import select`, and `sqlalchemy` is not available to the `galaxy.util` package; discovered in #16434).

There were 2 such instances in the util package:
1. The `get_user_by_username` function is moved as a standalone function into `managers.users`. (~~this is currently failing 2 tests -- I'm investigating~~)
2. The `dbkeys` module is moved into the `managers` package. 

    The data access code in `dbkeys` can be easily extracted into 2 functions which can be moved to `managers.hdas`. However, the calling code from `dbkeys` would still have to be moved out of the `util` package, otherwise it would have to import from `managers`. Is there a better way to rearrange this?

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: lib/galaxy/managers/dbkeys.py [hunk 1]
```diff
@@ -17,9 +21,9 @@
 
 def read_dbnames(filename):
     """Read build names from file"""
-    db_names = []
+    db_names: List = []
```
Comment (nsoranzo) [inline lib/galaxy/managers/dbkeys.py]: Not necessary, unless you have to specify the list content type.

File: lib/galaxy/managers/dbkeys.py [hunk 2]
```diff
@@ -17,9 +21,9 @@
...
     try:
-        ucsc_builds = {}
+        ucsc_builds: Dict = {}
```
Comment (nsoranzo) [inline lib/galaxy/managers/dbkeys.py]: As above.

File: lib/galaxy/managers/dbkeys.py (hunk 1 above)
Comment (jdavcs) [reply lib/galaxy/managers/dbkeys.py]:
```
(.venv) rivendell$ tox -e mypy
mypy: commands[0]> mypy test lib
lib/galaxy/managers/dbkeys.py:24: error: Need type annotation for "db_names" (hint: "db_names: List[<type>] = ...")  [var-annotated]
        db_names = []
        ^~~~~~~~
```

File: lib/galaxy/managers/dbkeys.py (hunk 2 above)
Comment (jdavcs) [reply lib/galaxy/managers/dbkeys.py]: same as above - mypy complains

File: lib/galaxy/managers/dbkeys.py (hunk 1 above)
Comment (nsoranzo) [reply lib/galaxy/managers/dbkeys.py]: These were due to:

```ini
[mypy-galaxy.util.dbkeys]
check_untyped_defs = False
```

in `mypy.ini` which we can now be removed.

Comment (jdavcs) [reply lib/galaxy/managers/dbkeys.py]: I've tried - same error as before. (I've cleared the cache) Does it work for you?

Comment (jdavcs) [reply lib/galaxy/managers/dbkeys.py]: I'm updating mypy.ini anyway, of course.

Comment (nsoranzo) [reply lib/galaxy/managers/dbkeys.py]: Thanks, I've improved the type annotation (and an import) in 2 commits I've just pushed to your branch, hope that's OK.

Comment (jdavcs) [reply lib/galaxy/managers/dbkeys.py]: of course, thanks!

**Pull request merged**